Release Notes for Java SDK 3.58.0 (#545)
SAP/cloud-sdk: docs/java/release-notes.mdx

  - (OData V2) _Service Order Template_ - `OP_API_SERVICE_ORDER_TEMPLATE_SRV_0001`
  - (OData V4) _Contract Accounting Business Transaction_ - `CABUSINESSPARTNER_0001`
  - (OData V4) _Handling Unit_ - `HANDLINGUNIT_0001`
  - (OData V4) _Maintenance Task List_ - `MAINTENANCETASKLIST_0001`
  - (OData V4) _Measuring Point_ - `MEASURINGPOINT_0001`
  - (OData V4) _Planned Order_ - `PLANNEDORDER_0001`
  - (OData V4) _Process Warehouse Task_ - `WAREHOUSEORDER_0001`
  - (OData V4) _Product Exclusion_ - `PRODUCTEXCLUSION_0001`
  - (OData V4) _Product Substitution_ - `PRODUCTSUBSTITUTION_0001`
  - (OData V4) _Supplier Item_ - `SUPPLIERITEM_0001`
  - (OData V4) _Warehouse Resource_ - `WAREHOUSERESOURCE_0001`

### Improvements

- Limit beans scanning to service classes only in dependencies `s4hana-api-odata`, `s4hana-api-odata-onpremise-2020` and`s4hana-api-odata-v4-onpremise-2020`.
  This significantly reduces startup times and memory consumption for TomEE applications that are missing their own `beans.xml` configuration.
  For consistency the beans scanning of `s4hana-api-odata-v4` is enabled now but also limited to service classes.

> - The naming of the default `ScpCfDestinationRetrievalStrategy` (`ALWAYS_SUBSCRIBER`) in the `ScpCfDestinationLoader` did not behave as the name implied, as it was taking the current tenant, which might, due to misconfiguration, also be the Provider Tenant.

Reviewer: ```suggestion
- The `ALWAYS_SUBSCRIBER` strategy of the default `ScpCfDestinationRetrievalStrategy` in the `ScpCfDestinationLoader` did not behave as its name implied.
It was taking the current tenant, which might also be a Provider Tenant due to misconfiguration.
```

>   This is now fixed by the following steps:

Reviewer: ```suggestion
  We fixed it by:
```

>   - Deprecation of the unclear `ALWAYS_SUBSCRIBER` strategy
>   - As a default providing a renamed alternative keeping the same behavior: `CURRENT_TENANT`
>   - Adding a new strategy `ONLY_SUBSCRIBER` which actively enforces that the current tenant is not the provider tenant

Reviewer: ```suggestion
  - Renaming `ALWAYS_SUBSCRIBER` to `CURRENT_TENANT` and making it a new default strategy.
  - Deprecation the confusingly named `ALWAYS_SUBSCRIBER` strategy
  - Adding a new `ONLY_SUBSCRIBER` strategy that actively enforces that a current tenant is not a provider tenant
```
